candle-core/src/display.rs

```diff
@@ -9,11 +9,20 @@ impl Tensor {
         &self,
         f: &mut std::fmt::Formatter,
     ) -> std::fmt::Result {
-        let prefix = match self.device() {
-            crate::Device::Cpu => "Cpu",
-            crate::Device::Cuda(_) => "Cuda",
+        #[cfg(feature = "cuda")]
```

Comment:
What about using `self.device().location()` instead, this should avoid the need for using the `cuda` feature gate so is more likely to work well when/if we add a new device some day?

Reply:
Indeed much cleaner! Thanks